specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -7892,6 +7892,14 @@
           "description": "The fabric specific object Id of the virtual machine.",
           "type": "string"
         },
+        "initialPrimaryFabricLocation": {
```

Comment:
add new properties in stable version better to be in a new api version.

Reply:
These new properties do not break any existing flows / contracts, we would like to handle within latest API version.

Reply:
These changes are considered a breaking change, a new API version is required. 

Take a look at 'New property added to response' for why this is required:
https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?rtime=WPZkgiIc10g

Reply:
@ryansbenson:
1) As per referred "_New property added to response_" in [RestAPI_BreakingChangeGuide](https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?rtime=WPZkgiIc10g), it is considered breaking if "_customer does a PUT using the model that was returned from the GET_", which does not apply for our GET / PUT ReplicationProtectedItem as property names are different (ref: [GET ReplicationProtectedItem](https://docs.microsoft.com/en-us/rest/api/site-recovery/replicationprotecteditems/get), [PUT ReplicationProtectedItem](https://docs.microsoft.com/en-us/rest/api/site-recovery/replicationprotecteditems/create)).
2) These newly added properties are option (not _required_ properties). As per PR #6380, it is required to bump up the API version only on adding _required_ properties. 
3) We were referring "[_Definition of a breaking change_](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change)" from "[_Microsoft REST API Guidelines_](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md)" - where it is said that "_Services MUST explicitly define their definition of a breaking change, especially with regard to adding new fields to JSON responses and adding new API arguments with default fields._" - In our case, we define the current changes as not breaking change.
4) Please share some best practices to be followed within service to implement, handle, and maintain multiple versions for every optional property we introduce (our service version handling is [here](https://msazure.visualstudio.com/One/_git/Azure-SiteRecovery?path=%2Fsrc%2FOnline%2FRecSrv%2FSRS%2FService%2FApi%2FService&version=GBdevelop)).

Reply:
IN section 12.3, https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change, it is explicitly stated -
For example, Azure defines the addition of a new JSON field in a response to be not backwards compatible.

Reply:
The guidance is to bundle all such changes so that every optional property addition does not require a new api-version.

Reply:
please mark this as readonly since its not settable